### PR TITLE
Apply patch to fix the-silversearcher

### PR DESCRIPTION
**Pull request by mattn**
the-silversearcher does not work when use pipe with external command on Windows.

```
C:\Temp> ag foo | vim -
```

This seems to be a bug of git that it break terminal sequences. This fix use "<NUL" to fix this behavior. (https://github.com/ggreer/the_silver_searcher/pull/1400)

### CODE REVIEW

**Comment by mattn** [comment]
Fixed some patches to be possible to apply for tip of the-silversearcher.

File: mingw-w64-ag/PKGBUILD [hunk 1]
```diff
@@ -17,22 +17,24 @@ depends=("${MINGW_PACKAGE_PREFIX}-pcre"
          "${MINGW_PACKAGE_PREFIX}-xz"
          "${MINGW_PACKAGE_PREFIX}-zlib")
 options=('staticlibs') # '!strip' 'debug')
-source=("${_realname}-${pkgver}.tar.gz"::"https://github.com/ggreer/the_silver_searcher/archive/${pkgver}.tar.gz"
+source=(${_longname}-${pkgver}::"git+https://github.com/ggreer/the_silver_searcher"
```
Comment (lazka) [inline mingw-w64-ag/PKGBUILD:20]: Can you revert this? We prefer releases.

**Comment by lazka** [comment]
needs a pkgrel bump

File: mingw-w64-ag/PKGBUILD (hunk 1 above)
Comment (mattn) [reply mingw-w64-ag/PKGBUILD:20]: the-silversearch does not tag yet to the latest fix.

Comment (mattn) [reply mingw-w64-ag/PKGBUILD:20]: https://github.com/ggreer/the_silver_searcher/commit/5a1c8d83ba1514ca8d045ba80403f93772fb371a

Comment (1480c1) [reply mingw-w64-ag/PKGBUILD:20]: If those fixes can be applied onto the latest patch, you could pull a patch from from the commit like https://github.com/ggreer/the_silver_searcher/commit/5a1c8d83ba1514ca8d045ba80403f93772fb371a.patch

Comment (lazka) [reply mingw-w64-ag/PKGBUILD:20]: Shouldn't your change also work with 2.2.0. The GCC 10 build issue should be solved with `CFLAGS+=' -fcommon'`

Comment (mattn) [reply mingw-w64-ag/PKGBUILD:20]: https://github.com/ggreer/the_silver_searcher/commit/21eaa1c4160b868b0c5bbf59da17974429f30055

This patch seems fixes CI-build.

**Comment by mattn** [comment]
Included patches to fix build.

**Comment by lazka** [comment]
thanks

**Comment by mattn** [comment]
Thank you too.

**Pull request merged**